File: backend/app/src/main/java/com/david/app/model/Evento.java
package com.david.app.model;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "eventos")
@Data
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class Evento {
    @Id
    private String id;

    private String titulo;
    private String ubicacion;
    private String clienteId;

    /**
     * Formato recomendado: YYYY-MM-DD (ej: 2026-03-15)
     */
    private String fecha;

    /**
     * Ej: "Evento", "Montaje", "Ensayo", "Otro"
     */
    private String tipo;

    /**
     * Ej: "blue", "orange", "green", "purple"
     */
    private String color;

    private String descripcion;
    private Double presupuesto;
    private Double presupuestoPresentado;
    private String presupuestoEstado; // Pendiente / Aceptado / Rechazado
    private Integer tecnicos;
    private List<EventoMaterial> materiales;
    private Integer dias;
    private Integer jornadas;
    private String modoCalculo;
    private List<EventoTecnico> tecnicosDetalle;
}

File: frontend/src/app/model/evento.ts
export interface Evento {
  id?: string;
  titulo: string;
  ubicacion?: string;
  clienteId?: string;
  fecha: string; // YYYY-MM-DD
  tipo: 'Evento' | 'Montaje' | 'Ensayo' | 'Otro';
  color: 'blue' | 'orange' | 'green' | 'purple';
  descripcion?: string;
  presupuesto?: number;
  presupuestoPresentado?: number;
  presupuestoEstado?: 'Pendiente' | 'Aceptado' | 'Rechazado';
  tecnicos?: number;
  materiales?: EventoMaterial[];
  dias?: number;
  jornadas?: number;
  modoCalculo?: 'Dias' | 'Jornadas';
  tecnicosDetalle?: EventoTecnico[];
}

export interface EventoMaterial {
  materialId?: string;
  nombre: string;
  cantidad: number;
}

export interface EventoTecnico {
  personalId?: string;
  nombre?: string;
  horas?: number;
  tarifaHora?: number;
}
